File: saransh_kunal-LINUX/Assignment/dotnet/C#/Interest_udf/Interest.cs
using System;

//Interest is a user-defined value type
struct Interest
{
    //instance fields
    private double principle;
    private float rate;
    private int period;

    //constructor
    
    public Interest(double principle, float rate, int period)
    {
        this.principle = principle;
        this.rate = rate;
        this.period = period;
    }

    public double SInterest()
    {
        double interest = (principle * rate * period) /100;
        return interest;
    }
}

File: saransh_kunal-LINUX/Codes/Foundations/Functions/interest.cpp
#include "interst.h"
#include <cmath>

long GetInterest(long principle, long rate, long period)
{
	if(principle<10000){
		throw principle;
	}
	
	long si;
	si = (principle * rate * period)/100;
	return si;

	//long amount;

}

